tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -811,271 +890,190 @@ def test_positive_ping(self):
-        # this auth mode. We then further restrict ourselves to the
-        # alphanumeric charset, because Foreman complains about incomplete
-        # multi-byte chars when latin1 chars are used.
+        # step 1: Create a new user with admin permissions
         login = gen_string('alphanumeric')
         password = gen_string('alphanumeric')
-
-        # step 1: Create a new user with admin permissions
         entities.User(admin=True, login=login, password=password).create()
 
         # step 2.1: Create a new organization
         server_config = get_nailgun_config()
         server_config.auth = (login, password)
         org = entities.Organization(server_config).create()
 
-        # step 2.2: Create 2 new lifecycle environments
+        # step 2.2: Clone and upload manifest
+        with manifests.clone() as manifest:
+            upload_manifest(org.id, manifest.content)
+
+        # step 2.3: Create a new lifecycle environment
         le1 = entities.LifecycleEnvironment(
             server_config,
             organization=org
         ).create()
-        le2 = entities.LifecycleEnvironment(
-            server_config,
-            organization=org,
-            prior=le1,
-        ).create()
 
-        # step 2.3: Create a custom product
+        # step 2.4: Create a custom product
         prod = entities.Product(server_config, organization=org).create()
 
-        # step 2.4: Create custom YUM repository
+        # step 2.5: Create custom YUM repository
         repo1 = entities.Repository(
             server_config,
             product=prod,
             content_type=u'yum',
             url=GOOGLE_CHROME_REPO
         ).create()
 
-        # step 2.5: Create custom PUPPET repository
+        # step 2.6: Create custom PUPPET repository
         repo2 = entities.Repository(
             server_config,
             product=prod,
             content_type=u'puppet',
             url=FAKE_0_PUPPET_REPO
         ).create()
 
-        # step 2.6: Synchronize both repositories
-        for repo in [repo1, repo2]:
+        # step 2.7: Enable a RedHat repository
+        repo3 = entities.Repository(id=enable_rhrepo_and_fetchid(
+            basearch='x86_64',
+            org_id=org.id,
+            product=PRDS['rhel'],
+            repo=REPOS['rhva6']['name'],
+            reposet=REPOSET['rhva6'],
+            releasever='6Server',
+        ))
+
+        # step 2.8: Synchronize the three repositories
+        for repo in [repo1, repo2, repo3]:
             repo.sync()
 
-        # step 2.7: Create content view
+        # step 2.9: Create content view
         content_view = entities.ContentView(
             server_config,
             organization=org
         ).create()
 
-        # step 2.8: Associate YUM repository to new content view
-        content_view.repository = [repo1]
+        # step 2.10: Associate the three repositories to new content view
+        content_view.repository = [repo1, repo3]
```

Comment:
Actually this is the step description and according to steps we have to create three repos and associate all of them. But since a puppet module have a different behavior we take care of it later.

I will update the steps to reflect the actual usage of the system.